unstract/task-abstraction/src/unstract/task_abstraction/workflow_bloated.py

```
@@ -0,0 +1,967 @@
+                    )
+                else:
+                    # Task without kwargs - pass input_data as positional arg
+                    result = self._execute_task_with_retry(step.task_name, input_data)
+
+                results.append(result)
+                logger.info(f"Parallel task '{step.task_name}' completed successfully")
+
+            except (TaskExecutionError, WorkflowTimeoutError) as e:
+                error_msg = f"Parallel task '{step.task_name}' failed: {e}"
+                logger.error(error_msg)
+                raise Exception(error_msg)
```

Ruff (0.13.1)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)